pymc3/distributions/posterior_predictive.py

```diff
@@ -0,0 +1,662 @@
+    from the posterior predictive distribution.  Notably it initializes the
+    ``_DrawValuesContext`` bookkeeping object and evaluates the "fast drawable"
+    parts of the model.'''
+        vars: List[Any] = self.vars
+        trace: _TraceDict = self.trace
+        samples: int = self.samples
+
+        # initialization phase
+        context = _DrawValuesContext.get_context()
+        assert isinstance(context, _DrawValuesContext)
+        with context:
+            drawn = context.drawn_vars
+            evaluated = {} # type: Dict[int, Any]
+            symbolic_params = []
+            for i, var in enumerate(vars):
+                if is_fast_drawable(var):
+                    evaluated[i] = self.draw_value(var)
+                    continue
+                name = getattr(var, 'name', None)
+                if (var, samples) in drawn:
+                    evaluated[i] = drawn[(var, samples)]
+                                # We filter out Deterministics by checking for `model` attribute
+                elif name is not None and hasattr(var, 'model') and name in trace.varnames:
+                    # param.name is in the trace.  Record it as drawn and evaluated
+                    drawn[(var, samples)] = evaluated[i] = trace[cast(str, name)]
+                else:
+                    # param still needs to be drawn
+                    symbolic_params.append((i, var))
+        self.evaluated = evaluated
+        self.symbolic_params = symbolic_params
+
+
+
+    def make_graph(self) -> None:
+        # Distribution parameters may be nodes which have named node-inputs
+        # specified in the point. Need to find the node-inputs, their
+        # parents and children to replace them.
+        symbolic_params = self.symbolic_params
+        self.leaf_nodes = {} # type: Dict[str, Any]
+        self.named_nodes_parents = {} # type: Dict[str, Any]
+        self.named_nodes_children = {} # type: Dict[str, Any]
+        for _, param in symbolic_params:
+            if hasattr(param, 'name'):
+                # Get the named nodes under the `param` node
+                nn, nnp, nnc = get_named_nodes_and_relations(param)
+                self.leaf_nodes.update(nn)
+                # Update the discovered parental relationships
+                for k in nnp.keys():
+                    if k not in self.named_nodes_parents.keys():
+                        self.named_nodes_parents[k] = nnp[k]
+                    else:
+                        self.named_nodes_parents[k].update(nnp[k])
+                # Update the discovered child relationships
+                for k in nnc.keys():
+                    if k not in self.named_nodes_children.keys():
+                        self.named_nodes_children[k] = nnc[k]
+                    else:
+                        self.named_nodes_children[k].update(nnc[k])
+
+    def draw_value(self, param, trace: Optional[_TraceDict]=None, givens = None):
```

Comment:
```suggestion
    def draw_value(self, param, trace: Optional[_TraceDict]=None, givens=None):
```

Reply:
Thanks for catching that: not sure why pylint didn't complain.